### ISSUE TITLE
Ef Core causes 'Access violation' when deleting Data that is tracked by ef core with ExecuteDeleteAsync

### ISSUE THREAD

**Issue by b-grimm-technoma** [dotnet/efcore#29743]
Using the new Method **ExecuteDeleteAsync** to bypass the Change Tracker in ef core to delete data that is tracked by EF Core causes an "Access violation".

1.
Data is being tracked by ef core and saved:
`await context.AddRangeAsync(wegpunkte);
await context.SaveChangesAsync();`

2.
Data, that has just been added, is deleted again with the **ExecuteDeleteAsync** method.
`await context.Set<Wegpunkt>().Where(predicate).ExecuteDeleteAsync();`

3.
New Data is added to Change Tracker
`await context.AddRangeAsync(wegpunkte);`
And the data is saved
`await context.SaveChangesAsync();`

After invoking the last **SaveChangesAsync** method the Program crashes with an Access violation Error that exits the process.
My assumption is, that the Change Tracker cannot find the previously tracked Entities that have been deleted with the **ExecuteDeleteAsync** Method.

EDIT:
A workaround exists by calling 
`context.ChangeTracker.Clear();`
right after calling the **ExecuteDeleteAsync** method.

**Comment by ajcvickers** [comment]
This issue is lacking enough information for us to be able to fully understand what is happening. Please attach a small, runnable project or post a small, runnable code listing that reproduces what you are seeing so that we can investigate.

**Comment by b-grimm-technoma** [comment]
I tried recreating this issue in a repo that should be able to run:
[https://github.com/b-grimm-technoma/EfCoreChangeTrackerTest.git](url)

The Program throws an Exception this time when calling the last SaveChanges method.

> InvalidOperationException: The connection does not support MultipleActiveResultSets.

Ef core probably believes that I have another open connection that has deleted the entities that it is tracking.
However the ChangeTracker does not detect a change when I deleted the entities with the ExecuteDelete method.
